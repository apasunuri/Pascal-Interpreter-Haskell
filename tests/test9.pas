program test9;

var
   gs, cgs, e, gi, ni : real;

function calculateGrossIncome(grossSales, costOfGoodsSold : real) : real;
begin
    calculateGrossIncome := grossSales - costOfGoodsSold;
end;

function calculateNetIncome(grossIncome, expenses : real) : real;
begin
    calculateNetIncome := grossIncome - expenses;
end; 

procedure produceIncomeStatement();
begin
    writeln('Gross Sales $');
    writeln(gs);
    writeln('Cost of Goods that were Sold $');
    writeln(cgs);
    writeln('Corporate Expenses $');
    writeln(e);
    gi := calculateGrossIncome(gs, cgs);
    ni := calculateNetIncome(gi, e);
    writeln('Gross Income $');
    writeln(gi);
    writeln('Net Income $');
    writeln(ni);
end;  

begin
    gs := 100000;
    cgs := 40000;
    e := 10000;
    produceIncomeStatement();
end.