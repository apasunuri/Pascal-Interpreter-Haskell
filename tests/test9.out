Gross Sales $
100000.0
Cost of Goods that were Sold $
40000.0
Corporate Expenses $
10000.0
Gross Income $
60000.0
Net Income $
50000.0